kv/notification_endpoint.go

```diff
@@ -212,6 +215,14 @@ func (s *Service) updateNotificationEndpoint(ctx context.Context, tx Tx, id infl
 	edp.SetCreatedAt(current.GetCRUDLog().CreatedAt)
 	edp.SetUpdatedAt(s.TimeGenerator.Now())
 
+	// TODO: delete old secrets?
+	for _, fld := range edp.SecretFields() {
+		if fld.Value != nil {
+			if err := s.putSecret(ctx, tx, edp.GetOrgID(), fld.Key, *fld.Value); err != nil {
```

Comment:
the secret service is through vault via tcp, not kv. So this won't work at all, and will break the cloud

Reply:
Riight. I completely forgot that's why this is going through http instead of kv.

I still feel like this could be a smell, but I'd have to think about how we could refactor it.

Reply:
regardless of where the secrets are, should still be dep injected into the notification service somewhere. The lack of service definition in the codebase is super painful at times like these. The current abstraction leaks like a sieve 😢